src/node/txpackagetracker.cpp

```diff
@@ -195,36 +213,45 @@ class TxPackageTracker::Impl {
         }
         std::vector<GenTxid> results;
         for (const auto& gtxid : tracker_requestable) {
-            LogPrint(BCLog::TXPACKAGES, "\nResolving orphan %s, requesting by txids of parents from peer=%d\n", gtxid.GetHash().ToString(), nodeid);
-            const auto ptx = m_orphanage.GetTx(gtxid.GetHash());
-            if (!ptx) {
-                // We can't request ancpkginfo and we have no way of knowing what the missing
-                // parents are (it could also be that the orphan has already been resolved).
-                // Give up.
-                orphan_request_tracker.ForgetTxHash(gtxid.GetHash());
-                LogPrint(BCLog::TXPACKAGES, "\nForgetting orphan %s from peer=%d\n", gtxid.GetHash().ToString(), nodeid);
-                continue;
+            if (gtxid.IsWtxid()) {
+                Assume(info_per_peer.find(nodeid) != info_per_peer.end());
+                // Add the orphan's wtxid as-is.
+                LogPrint(BCLog::TXPACKAGES, "\nResolving orphan %s, requesting by ancpkginfo from peer=%d\n", gtxid.GetHash().ToString(), nodeid);
+                results.emplace_back(gtxid);
```

Comment:
I think that the splitting into commits is not ideal here: In ae2664793b1d0d86bfa32c3e635d83b1ee083e60, we add logic to return the wtxid, expecting it to be requested by `MSG_ANCPKGINFO`. But the net_processing code to actually do this is only added in dd6fb139b8887f3400a65dc0d4c4ee54cf53d0bc, so that at ae2664793b1d0d86bfa32c3e635d83b1ee083e60, we'd send out a `MSG_TX` getdata, thus re-requesting a TX we already know. These two things should go together into one commit.